ui/src/app/views/concept-set-list/component.tsx

```diff
@@ -0,0 +1,123 @@
+import {Component} from '@angular/core';
+import * as React from 'react';
+
+import {
+  DomainInfo,
+  RecentResource,
+  WorkspaceAccessLevel,
+} from 'generated/fetch';
+
+import {conceptsApi, conceptSetsApi} from 'app/services/swagger-fetch-clients';
+import {reactStyles, ReactWrapperBase, withCurrentWorkspace} from 'app/utils/index';
+
+import {CreateConceptSetModal} from 'app/views/conceptset-create-modal/component';
+
+import {CardButton} from 'app/components/buttons';
+import {ClrIcon} from 'app/components/icons';
+import {TooltipTrigger} from 'app/components/popups';
+import {SpinnerOverlay} from 'app/components/spinners';
+import {WorkspaceData} from 'app/services/workspace-storage.service';
+import {convertToResources, ResourceType} from 'app/utils/resourceActions';
+import {ResourceCard} from 'app/views/resource-card/component';
+
+const styles = reactStyles({
+  pageArea: {
+    display: 'flex',
+    justifyContent: 'flex-start'
+  },
+  resourceCardArea: {
+    display: 'flex',
+    flexWrap: 'wrap'
+  }
+});
+
+export const ConceptSetsList = withCurrentWorkspace()(
+  class extends React.Component<{workspace: WorkspaceData},
+      {conceptSetsList: RecentResource[], conceptSetsLoading: boolean,
+        conceptDomainsList: Array<DomainInfo>, createModalOpen: boolean}> {
+    constructor(props) {
+      super(props);
+      this.state = {
+        conceptSetsList: undefined,
+        conceptSetsLoading: true,
+        conceptDomainsList: undefined,
+        createModalOpen: false
+      };
+    }
+
+    componentDidMount() {
+      try {
+        this.getDomainInfo();
+      } finally {
+        this.loadConceptSets();
+      }
+    }
+
+    async getDomainInfo() {
+      const {namespace, id} = this.props.workspace;
+      try {
+        const resp = await conceptsApi().getDomainInfo(namespace, id);
+        this.setState({conceptDomainsList: resp.items});
+      } catch (error) {
+        console.log(error);
+      }
+    }
+
+    async loadConceptSets() {
+      const {namespace, id, accessLevel} = this.props.workspace;
+      try {
+        const resp = await conceptSetsApi().getConceptSetsInWorkspace(namespace, id);
+        const conceptSets = convertToResources(resp.items, namespace, id,
+          accessLevel as unknown as WorkspaceAccessLevel, ResourceType.CONCEPT_SET);
+
+        this.setState({conceptSetsList: conceptSets, conceptSetsLoading: false});
+      } catch (error) {
+        console.log(error);
+      }
+    }
+
+    render() {
+      const {conceptSetsLoading, conceptSetsList, conceptDomainsList, createModalOpen} = this.state;
+      const accessLevel = this.props.workspace.accessLevel as unknown as WorkspaceAccessLevel;
```

Comment:
we actually had to do this because during the react conversion some objects were created in the generated/fetch api code and others just in the generated code.  so although they were effectively the same object, it would cause a type mismatch.  we should be able to get rid of all these castings when the conversion is done. probably want to file a ticket for that.

Reply:
actually if there is a react cleanup ticket i can add it to the tasks for that.

Reply:
added to this ticket: https://precisionmedicineinitiative.atlassian.net/browse/RW-2013

Reply:
Thanks for adding. I'm happy to hear it's a temporary situation!